src/config.c

```diff
@@ -24,12 +25,15 @@
 
 typedef enum {
     NM_CONFIG_TYPE_MENU_ITEM = 1,
+    NM_CONFIG_TYPE_GENERATOR = 2,
 } nm_config_type_t;
 
 struct nm_config_t {
     nm_config_type_t type;
+    bool generated;
```

Comment:
We're relying on `calloc` to make sure this defaults to `false`, right?

Reply:
Yep (we also rely on it to make things zero in a few other places).